vendor/bat-native-ledger/src/bat_helper.cc

```diff
@@ -2618,6 +2619,24 @@ uint64_t getMediaDuration(const std::map<std::string, std::string>& data,
   return duration;
 }
 
+bool HasSameDomainAndPath(
+    const std::string& url_to_validate,
+    const std::string& domain_to_match,
+    const std::string& path_to_match) {
+  GURL gurl(url_to_validate);
+  GURL gurl_to_match(domain_to_match);
+  if (gurl.is_valid() && gurl_to_match.is_valid() &&
+      gurl.DomainIs(gurl_to_match.spec()) &&
+      gurl.has_path() && gurl_to_match.has_path() &&
+      gurl.path().find(gurl_to_match.path())) {
+    return true;
+  }
+  return gurl.is_valid() &&
+         gurl.host().find(domain_to_match) != std::string::npos &&
+         gurl.path().find(path_to_match) != std::string::npos &&
+         !domain_to_match.empty();
```

Comment:
I guess i'm overall confused about the logic in here. i think all you need in this entire function is to check:

1. `gurl.is_valid()`, `domain_to_match` and `path_to_match` are non-empty
2. `gurl.DomainIs(domain_to_match)`
3. `gurl.path()` starts with `path_to_match`

if so return true; otherwise return false

Reply:
Agree. Originally didn't have path parameter and that second half should have been refactored out. Fixed